### PR TITLE
Add GitHub Actions Tests Run

### PR DESCRIPTION
**Pull request by cooperlees**
- Lets run Unit Tests on PRs + pushes
- Use unittests __main__.py directly as `python setup.py test` is deprecated

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      run: |
+        python -m pip install --upgrade pip setuptools
+
+    - name: Install latest version of memory_analyzer
+      run: |
+        python -m pip install .
+
+    - name: Run Unitests
+      run: |
+        python memory_analyzer/tests/__main__.py
```
Comment (thatch) [inline .github/workflows/tests.yml]: something like `python -m unittest memory_analyzer.tests` (or `make test`)?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Building ptr for python ${{ matrix.python-version }} on ${{matrix.os}}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7]
```
Comment (thatch) [inline .github/workflows/tests.yml]: pretty sure 3.8 is available now, is there a reason to exclude?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    - uses: actions/checkout@v1
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Update pip + setuptools
+      run: |
+        python -m pip install --upgrade pip setuptools
```
Comment (thatch) [inline .github/workflows/tests.yml]: I think the most recent copy of this action is https://github.com/jreese/aioitertools/blob/master/.github/workflows/build.yml if you want something to crib.  Using `make setup` will obey whatever requirements/requrements-dev that exists in the repo.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: memory_analyzer_tests
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Building ptr for python ${{ matrix.python-version }} on ${{matrix.os}}
```
Comment (thatch) [inline .github/workflows/tests.yml]: s/ptr/memory-analyzer/

**Comment by cooperlees** [comment]
Everything else I ask you to do in follow up PRs. Thanks for letting me know 3.8 is available now!

**Comment by lisroach** [comment]
Looks good to me, thanks for putting this together @cooperlees!

**Pull request merged**